### PR TITLE
Add RHE Linux 8 stream and AWS linux to software requirement doc

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Add RHE Linux 8 stream and AWS linux to software requirement doc

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ TiDB 作为一款开源分布式 NewSQL 数据库，可以很好地部署和运
     |  操作系统   |   支持的 CPU 架构   |
     |   :---   |   :---   |
     | <ul><li>Red Hat Enterprise Linux 8.4 及以上的 8.x 版本</li><li>CentOS 8.4 及以上的 8.x 版本</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
+    | Red Hat Enterprise Linux 8 Stream |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:21]: @dveeden PTAL. Adding the Red Hat Enterprise Linux 8 Stream to TiDB-supported Linux OS.

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -18,7 +18,9 @@ TiDB 作为一款开源分布式 NewSQL 数据库，可以很好地部署和运
...
     | <ul><li>Red Hat Enterprise Linux 7.3 及以上的 7.x 版本</li><li>CentOS 7.3 及以上的 7.x 版本</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
+    |  AWS Linux         |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:23]: @easonn7 PTAL

File: hardware-and-software-requirements.md (hunk 1 above)
Comment (dveeden) [reply hardware-and-software-requirements.md:21]: Is this RHEL 8 or CentOS 8 Steam? "Red Hat Enterprise Linux 8 Stream" doesn't exist. I assume this is CentOS 8 Stream.

```suggestion
    | CentOS 8 Stream |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
```

File: hardware-and-software-requirements.md (hunk 2 above)
Comment (dveeden) [reply hardware-and-software-requirements.md:23]:
```suggestion
    |  Amazon Linux 2         |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
```

- "AWS Linux" doesn't exist. "Amazon Linux" does.
- For "Amazon Linux" we should only support version 2 as there are huge differences between Amazon Linux 2 and Amazon Linux 2022. While Amazon Linux 2 is compatible with RHEL7/CentOS7, Amazon Linux 2022 is built with Fedora Linux as a base, but with many big enhancements on top of that.

References:

- https://aws.amazon.com/linux/amazon-linux-2022/
- https://aws.amazon.com/amazon-linux-2/
- https://docs.aws.amazon.com/linux/al2022/ug/compare-al2-to-AL2022.html

**Review by dveeden — CHANGES_REQUESTED** [review]
Please see the comments on this PR and the comments on the linked PR.

**Comment by easonn7** [comment]
LGTM

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -18,7 +18,9 @@ TiDB 作为一款开源分布式 NewSQL 数据库，可以很好地部署和运
     |  操作系统   |   支持的 CPU 架构   |
     |   :---   |   :---   |
     | <ul><li>Red Hat Enterprise Linux 8.4 及以上的 8.x 版本</li><li>CentOS 8.4 及以上的 8.x 版本</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:20]:
```suggestion
    | Red Hat Enterprise Linux 8.4 及以上的 8.x 版本  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
```

**Comment by JasonWu0506** [comment]
LGTM

**Review by dveeden — APPROVED** [review]

**Pull request merged**